core/metrics/exporter.go

```diff
@@ -0,0 +1,223 @@
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/jhunt/go-log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/shieldproject/shield/core/bus"
+)
+
+type Config struct {
+	Namespace        string
+	TenantCount      int
+	AgentCount       int
+	TargetCount      int
+	StoreCount       int
+	JobCount         int
+	TaskCount        int
+	ArchiveCount     int
+	StorageUsedCount int64
+}
+
+type Metrics struct {
+	bus                   *bus.Bus
+	tenantsGauge          prometheus.Gauge
+	agentsGauge           prometheus.Gauge
+	targetsGauge          prometheus.Gauge
+	storesGauge           prometheus.Gauge
+	jobsGauge             prometheus.Gauge
+	tasksGauge            prometheus.Gauge
+	archivesGauge         prometheus.Gauge
+	storageUsedBytesGauge prometheus.Gauge
+}
+
+const (
+	tenantsTotal      = "tenants_total"
+	agentsTotal       = "agents_total"
+	targetsTotal      = "targets_total"
+	storesTotal       = "stores_total"
+	jobsTotal         = "jobs_total"
+	tasksTotal        = "tasks_total"
+	archivesTotal     = "archives_total"
+	storageBytesTotal = "storage_used_bytes"
+	/* TODO
+	targetHealthStatus = "target_health_status"
+	storeHealthStatus  = "storeHealthStatus"
+	coreStatus         = "core_status" */
+)
+
+func New(config Config) *Metrics {
+	metrics := &Metrics{}
+	namespace := config.Namespace
+	metrics.tenantsGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      tenantsTotal,
+			Help:      "How many Tenants exist",
+		})
+
+	metrics.agentsGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      agentsTotal,
+			Help:      "How many SHIELD Agents have been registered",
+		})
+
+	metrics.targetsGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      targetsTotal,
+			Help:      "How many Target Systems have been defined",
+		})
+
+	metrics.storesGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      storesTotal,
+			Help:      "How many Cloud Storage Systems have been defined",
+		})
+
+	metrics.jobsGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      jobsTotal,
+			Help:      "How many backup jobs have been defined",
+		})
+
+	metrics.tasksGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      tasksTotal,
+			Help:      "How many tasks have been created",
+		})
+
+	metrics.archivesGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      archivesTotal,
+			Help:      "How many Backup Archives have been generated",
+		})
+
+	metrics.storageUsedBytesGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      storageBytesTotal,
+			Help:      "How much storage has been used, in bytes.",
+		})
+
+	prometheus.MustRegister(metrics.tenantsGauge)
```

Comment:
MustRegister takes a variadic list. You can provide all of this registration as one call to `MustRegister`.